### PR TITLE
Modify rule S2638: LaYC format

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2638/java/rule.adoc [hunk 1]
```diff
@@ -1,11 +1,14 @@
+This rule raises an issue when a overriding method changing a contract defined in the superclass.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S2638/java/rule.adoc:1]: This rule raises an issue when an overriding method changes a contract defined in a superclass.

File: rules/S2638/python/rule.adoc [hunk 2]
```diff
@@ -1,3 +1,5 @@
+This rule raises an issue when a overriding method changing a contract defined in the superclass.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S2638/python/rule.adoc:1]: Same here:
This rule raises an issue when an overriding method changes a contract defined in a superclass.

File: rules/S2638/python/rule.adoc [hunk 3]
```diff
@@ -26,14 +28,33 @@ The following modifications are not ok:
 
 This rule raises an issue when the signature of an overriding method does not accept the same parameters as the overriden one. Only instance methods are considered, class methods and static methods are ignored.
 
-=== Noncompliant code example
+=== Exceptions
+
+In theory renaming parameters also breaks Liskov Substitution Principle. Arguments can't be passed via keyword arguments anymore. However, https://www.python.org/dev/peps/pep-0570/#consistency-in-subclasses[as PEP-570 says], it is common to rename parameters when it improves code readability and when arguments are always passed by position.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S2638/python/rule.adoc:33]: I'd add a comma:
In theory, ...
I'd also reword a bit this segment:
However, PEP-570 indicates it is common ...

File: rules/S2638/python/rule.adoc [hunk 4]
```diff
@@ -26,14 +28,33 @@ The following modifications are not ok:
...
+
+"Positional-Only Parameters" were introduced in python 3.8 to solve this problem. As most programs will need to support older versions of python, this rule won't raise an issue on renamed parameters.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S2638/python/rule.adoc:35]: Capitalization: Python

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Some minor suggestions

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM

**Review by jean-jimbo-sonarsource — CHANGES_REQUESTED** [review]
Please capitalize "ok" to OK, as technically _okay_ is the full word version of _OK_.
Also, a link to the Liskov Substitution Principle under "Why is this an issue?" is helpful rather than hoping to find it in the resources section.

**Review by jean-jimbo-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**